### PR TITLE
Stabilize most commonly failing tests

### LINKED ISSUE #36666: java - core - ci flaky tests: attestation
Example build logs: https://dev.azure.com/azure-sdk/public/_build/results?buildId=3061428&view=logs&j=d5b2858d-8026-507f-519e-246ad3aed708&t=bc183069-b5ec-5c0d-742d-852f48003af5&s=ff05ad62-bb9a-53b6-ce9f-72f329a63e7c 

This failure happened in Attempt # 2. 

Error message at the end of the build: 

```
2023-09-06T21:29:06.2495883Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.1.0:test (default-test) on project azure-security-attestation: There are test failures.
2023-09-06T21:29:06.2496548Z [ERROR] 
2023-09-06T21:29:06.2496901Z [ERROR] Please refer to /mnt/vss/_work/1/s/sdk/attestation/azure-security-attestation/target/surefire-reports for the individual test results.
2023-09-06T21:29:06.2497272Z [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
2023-09-06T21:29:06.2497492Z [ERROR] -> [Help 1]
2023-09-06T21:29:06.2497647Z [ERROR] 
2023-09-06T21:29:06.2528912Z [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2023-09-06T21:29:06.2529720Z [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2023-09-06T21:29:06.2530118Z [ERROR] 
2023-09-06T21:29:06.2530538Z [ERROR] For more information about the errors and possible solutions, please read the following articles:
2023-09-06T21:29:06.2530968Z [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
2023-09-06T21:29:06.2531312Z [ERROR] 
2023-09-06T21:29:06.2531644Z [ERROR] After correcting the problems, you can resume the build with the command
2023-09-06T21:29:06.2532112Z [ERROR]   mvn <args> -rf :azure-security-attestation
2023-09-06T21:29:06.4187399Z The process '/usr/bin/mvn' failed with exit code 1
```

Suspected failure information: 

```
2023-09-06T21:12:35.9773325Z [ERROR] Tests run: 15, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.533 s <<< FAILURE! - in com.azure.security.attestation.models.AttestationTokenTests
2023-09-06T21:12:35.9848945Z [ERROR] com.azure.security.attestation.models.AttestationTokenTests.verifyAttestationTokenExpireTimeout  Time elapsed: 0.261 s  <<< FAILURE!
2023-09-06T21:12:35.9849956Z org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
2023-09-06T21:12:35.9851251Z 	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
2023-09-06T21:12:35.9852106Z 	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
2023-09-06T21:12:35.9852869Z 	at org.junit.jupiter.api.AssertTrue.failNotTrue(AssertTrue.java:63)
2023-09-06T21:12:35.9853611Z 	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:36)
2023-09-06T21:12:35.9854225Z 	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:31)
2023-09-06T21:12:35.9854685Z 	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:180)
2023-09-06T21:12:35.9855552Z 	at com.azure.security.attestation@1.2.0-beta.1/com.azure.security.attestation.models.AttestationTokenTests.verifyAttestationTokenExpireTimeout(AttestationTokenTests.java:299)
2023-09-06T21:12:35.9856069Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-09-06T21:12:35.9856450Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-09-06T21:12:35.9856852Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-09-06T21:12:35.9857233Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-09-06T21:12:35.9857604Z 	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
2023-09-06T21:12:35.9858013Z 	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
2023-09-06T21:12:35.9858499Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
2023-09-06T21:12:35.9859040Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
2023-09-06T21:12:35.9859526Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
2023-09-06T21:12:35.9860286Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
2023-09-06T21:12:35.9860831Z 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
2023-09-06T21:12:35.9861362Z 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
2023-09-06T21:12:35.9861777Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
2023-09-06T21:12:35.9862222Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
2023-09-06T21:12:35.9862678Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
2023-09-06T21:12:35.9863115Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
2023-09-06T21:12:35.9863526Z 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
2023-09-06T21:12:35.9863947Z 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
2023-09-06T21:12:35.9864403Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
2023-09-06T21:12:35.9864883Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-09-06T21:12:35.9865356Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
2023-09-06T21:12:35.9865841Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
2023-09-06T21:12:35.9866236Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
2023-09-06T21:12:35.9866900Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
2023-09-06T21:12:35.9867352Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-09-06T21:12:35.9867736Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2023-09-06T21:12:35.9868147Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2023-09-06T21:12:35.9868607Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2023-09-06T21:12:35.9869082Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-09-06T21:12:35.9869555Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2023-09-06T21:12:35.9870002Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2023-09-06T21:12:35.9870532Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:202)
2023-09-06T21:12:35.9871168Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:172)
2023-09-06T21:12:35.9871742Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:152)
2023-09-06T21:12:35.9872305Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2023-09-06T21:12:35.9872780Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-09-06T21:12:35.9873459Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2023-09-06T21:12:35.9873911Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2023-09-06T21:12:35.9874335Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2023-09-06T21:12:35.9874849Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-09-06T21:12:35.9875276Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2023-09-06T21:12:35.9875647Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2023-09-06T21:12:35.9876101Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:202)
2023-09-06T21:12:35.9876589Z 	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
2023-09-06T21:12:35.9876970Z 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
2023-09-06T21:12:35.9877354Z 	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:436)
2023-09-06T21:12:35.9877717Z 	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:670)
2023-09-06T21:12:35.9878215Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.joinConcurrentTasksInReverseOrderToEnableWorkStealing(ForkJoinPoolHierarchicalTestExecutorService.java:179)
2023-09-06T21:12:35.9878926Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:153)
2023-09-06T21:12:35.9879483Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2023-09-06T21:12:35.9879984Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-09-06T21:12:35.9880611Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2023-09-06T21:12:35.9881076Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2023-09-06T21:12:35.9881517Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2023-09-06T21:12:35.9881976Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-09-06T21:12:35.9882397Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2023-09-06T21:12:35.9882801Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2023-09-06T21:12:35.9883270Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:202)
2023-09-06T21:12:35.9883786Z 	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
2023-09-06T21:12:35.9884156Z 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
2023-09-06T21:12:35.9884510Z 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
2023-09-06T21:12:35.9884908Z 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
2023-09-06T21:12:35.9885306Z 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
2023-09-06T21:12:35.9885717Z 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
2023-09-06T21:12:35.9885949Z 
```

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Fixes #36666

Attempts to stabilize the most commonly failing tests in the Core pipeline.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java [hunk 1]
```diff
@@ -132,11 +137,18 @@ private Mono<HttpResponse> attemptAsync(HttpRequest request, boolean eagerlyRead
...
+            .flatMap(responseAndHeaders -> {
+                HttpResponse response = responseAndHeaders.getT1();
                 if (addProxyHandler && response.getStatusCode() == 407) {
-                    return proxyRetry
-                        ? Mono.error(new ProxyConnectException("Connection to proxy failed."))
-                        : Mono.error(new ProxyConnectException("First attempt to connect to proxy failed."));
+                    if (proxyRetry) {
+                        // Exhausted retry attempt return an error.
+                        return Mono.error(new HttpProxyHandler.HttpProxyConnectException(
+                            "Failed to connect to proxy. Status: 407", responseAndHeaders.getT2()));
```
Comment (lmolkova) [inline sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java:146]: Curious why we can't use `response.getHeaders()` here?

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java [hunk 2]
```diff
@@ -59,14 +59,20 @@ public DefaultLogger(Class<?> clazz) {
...
      */
     public DefaultLogger(String className) {
-        String classPath;
+        this(getClassPathFromClassName(className), false);
+    }
+
+    private static String getClassPathFromClassName(String className) {
         try {
-            classPath = Class.forName(className).getCanonicalName();
+            return Class.forName(className).getCanonicalName();
```
Comment (lmolkova) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java:67]: nit: maybe call overload with `initialize` flag set to false to minimize side-effects?

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java [hunk 3]
```diff
@@ -377,9 +382,57 @@ private void log(String levelName, String message, Throwable t) {
...
-    private String getFormattedDate() {
+    private static String getFormattedDate() {
         LocalDateTime now = LocalDateTime.now();
-        return DATE_FORMAT.format(now);
+
+        // yyyy-MM-dd HH:mm:ss.SSS
+        // 23 characters that will be ASCII
+        byte[] bytes = new byte[23];
+
+        // yyyy-
```
Comment (lmolkova) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java:392]: year `10000` will be a tough one 🤣

**Review by lmolkova — APPROVED** [review]

File: sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java:146]: This takes the Netty `HttpHeaders` object, not the azure-core `HttpHeaders`.

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java:392]: I'll be worried when I see that day 😄

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java:67]: I'm a bit confused on this comment, this is a utility method to help the `String`-based constructor as `this` has to be the first statement in a constructor.

**Pull request merged**